measures/ApplyUpgrade/measure.rb

```diff
@@ -338,8 +339,43 @@ def run(model, runner, user_arguments)
     end
     measures['BuildResidentialHPXML'][0]['additional_properties'] = additional_properties.join('|') unless additional_properties.empty?
 
-    # Retain HVAC capacities
+    # Heat Pump Backup
+    heat_pump_backup_type = measures['ResStockArguments'][0]['heat_pump_backup_type']
+    if heat_pump_backup_type == 'existing'
+      heating_system = get_heating_system(hpxml)
+
+      if not heating_system.nil?
+        if [HPXML::HVACTypeFurnace, HPXML::HVACTypeFixedHeater].include?(heating_system.heating_system_type) # Integrated
```

Comment:
This feels pretty brittle/risky. It'll be easy for this code to not get updated if new heating properties or system types are added.

I haven't thought about it as hard as you have, but I don't really understand the reason for choosing this approach. Why not keep things how they were but simply preserve the heating capacity when "separate" is chosen and there is an existing system?

Reply:
Furnace, for example, is not a valid "second heating system".

Reply:
Thinking about it more, I realize part of the motivation is to allow preserving not just capacity but also efficiency/fueltype. So you'd be able to specify a single heat pump upgrade that applies to many different existing fossil fuel systems. I just worry a bit about all the code that is being introduced.

Reply:
> Furnace, for example, is not a valid "second heating system".

If the furnace is backup to a central ASHP, then it is allowed (as an integrated backup system).

If the furnace is backup to a ductless mini-split, then I agree it's not currently allowed. I would suggest we update BuildResHPXML to allow it. So long as we only end up with one duct system for the home, I think it should be straightforward.

Reply:
> > Furnace, for example, is not a valid "second heating system".
> 
> If the furnace is backup to a central ASHP, then it is allowed (as an integrated backup system).
> 
> If the furnace is backup to a ductless mini-split, then I agree it's not currently allowed. I would suggest we update BuildResHPXML to allow it. So long as we only end up with one duct system for the home, I think it should be straightforward.

By "second heating system", I meant it's not allowed here: https://github.com/NREL/OpenStudio-HPXML/blob/master/BuildResidentialHPXML/measure.rb#L1340-L1354.

Reply:
> By "second heating system", I meant it's not allowed here: https://github.com/NREL/OpenStudio-HPXML/blob/master/BuildResidentialHPXML/measure.rb#L1340-L1354.

Yeah, I understand. Okay, what I said earlier wasn't quite right, let me try that again.

We should update the BuildResHPXML measure to allow a furnace as a second heating system -- so long as the home doesn't end up with two duct systems. 
1. If there is a ducted heat pump w/ backup furnace, the backup type needs to be "integrated". (If it was "separate", that would imply two duct systems. OS-HPXML can handle that, but that's too much complexity for the BuildResHPXML measure and would be a _very_ uncommon situation.)
2. If there is a ductless heat pump w/ backup furnace, the backup type needs to be separate. (A ductless heat pump can't have ducted integrated backup. I wonder what OS-HPXML does in that situation...)